By modifying pll_base to match different HSE values

qpzm1258 opened the pull request:
By modifying pll_base to match different HSE values, the SYSCLK can be kept constant at 400MHz. 

HSE 8mhz->pll_base = 8Mhz
HSE 12mhz,16mhz,20mzh,24mhz->pll_base = 4Mh
HSE 25mhz->pll_base = 5Mhz

Signed-off-by: 笑饮砒霜 <qpzm1258@163.com>

qpzm1258 committed:
By modifying pll_base to match different HSE values, the SYSCLK can be kept constant at 400MHz.

By modifying pll_base to match different HSE values, the SYSCLK can be kept constant at 400MHz.
HSE 8mhz->pll_base = 8Mhz
HSE 12mhz,16mhz,20mzh,24mhz->pll_base = 4Mh
HSE 25mhz->pll_base = 5Mhz

qpzm1258 commented:
It seems that 8Mhz HSE can also use 4Mhz pll_base.

nefelim4ag commented:
I see no issues with the change itself, but per the contribution guidelines
https://www.klipper3d.org/CONTRIBUTING.html#what-to-expect-in-a-review
Each commit requires a Signed-off-by line with a real name.

_Also, please, do not recreate PR 3rd time. There is a force push for that, or just add commits on top, which could be squashed._

Is there a real H7 board that will benefit from that?

qpzm1258 commented:
> I see no issues with the change itself, but per the contribution guidelines https://www.klipper3d.org/CONTRIBUTING.html#what-to-expect-in-a-review Each commit requires a Signed-off-by line with a real name.
> 
> _Also, please, do not recreate PR 3rd time. There is a force push for that, or just add commits on top, which could be squashed._
> 
> Is there a real H7 board that will benefit from that?

I'm sorry I haven't read this document before. I don't know the relevant rules.
I made a board based on stm32h750vbt6. When I use 8Mhz HSE. It works at 64Mhz.Usb serial port keeps reconnecting. So I modified the source code. I tested 8Mhz HSE and 24Mhz HSE.It work fine.

nefelim4ag reviewed (approved):
![image](https://github.com/user-attachments/assets/f18ad99e-7240-4bc0-9e6b-c340f5fa7372)
![image](https://github.com/user-attachments/assets/950b6cf8-6605-4e43-9083-fbc573db6a6a)
![image](https://github.com/user-attachments/assets/b9ae41c9-8f76-4b07-8451-dc0ed20ce62e)

lgtm

qpzm1258 committed:
Modify pll_base to 4M with HSE 8Mhz 

Through testing, it has been confirmed that when HSE is 8MHz, pll_base at 4MHz can also work properly.

KevinOConnor commented:
Thanks.  I made this change in a slightly different way (commit 5001983d).

-Kevin
